packages/bundler-plugin-core/src/utils/normalizeOptions.ts

```diff
@@ -0,0 +1,72 @@
+import { type Options } from "../types.ts";
+import { z } from "zod";
+
+export type NormalizedOptions = z.infer<typeof optionsSchema> & Options;
+
+const validBundleName = /^[\w\d_:/@\.{}\[\]$-]+$/;
+
+const optionsSchema = z.object({
+  apiUrl: z.string().url().default("https://api.codecov.io"),
+  dryRun: z.boolean().default(false),
+  bundleName: z.string().regex(validBundleName),
```

Comment:
we should just validate all the options here, add `retryCount` (and default to 3) and others.

Really helps the error cases when people are not using TS.

It means we don't have to spread userOptions here either.